File: PythonApplication1/Adapter.py
from abc import ABCMeta, abstractmethod

class ITarget(metaclass=ABCMeta):
    @abstractmethod
    def request(self):
        pass

class Adaptee:
    def operation(self):
        pass

class ObjectAdapter(ITarget):
    def __init__(self):
        self._adaptee = Adaptee()
    
    def request(self):
        self._adaptee.operation();

class ClassAdapter(Adaptee, ITarget):
    def __init__(self):
        self._operation = self.operation

    def __getattribute__(self, name):
        if name == "operation":
            raise AttributeError("'ClassAdapter' object has no attribute '{}'".format(name));
        return super().__getattribute__(name)
        

    def request(self):
        self._operation()

target = ObjectAdapter()
target.request()

target2 = ClassAdapter()
target2.request()
target2.operation()

File: ConsoleApplication1/Adapter.cs
using System;

interface ITarget { void Request(); }

class Client
{
    static void Main()
    {
        var target = new ObjectAdapter();
        target.Request();

        var target2 = ClassAdapter.Create();
        
    }
}

class ClassAdapter : Adaptee, ITarget
{
    private ClassAdapter() { }
    public void Request() => Operation();
    public static ITarget Create() => new ClassAdapter();
}

class ObjectAdapter : ITarget
{
    Adaptee adaptee = new Adaptee();
    public void Request() => adaptee.Operation();
}

class Adaptee
{
    public void Operation() => Console.WriteLine("Adaptee.Operation() called");
}